open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineSchemaTypeHandler.java

```diff
@@ -231,96 +239,130 @@ public void removeSchemaType(String userId, String schemaTypeGUID) throws Invali
-            SchemaAttribute schemaAttribute = createTabularColumn(attribute);
 
-            Optional<EntityDetail> schemaAttributeEntity = findSchemaAttributeEntity(userId, schemaAttribute.getQualifiedName());
+            Optional<EntityDetail> schemaAttributeEntity = findSchemaAttributeEntity(userId, attribute.getQualifiedName());
 
             if (!schemaAttributeEntity.isPresent()) {
-                createSchemaAttribute(userId, schemaTypeGUID, schemaAttribute, attribute.getDataType(), externalSourceName);
+                createSchemaAttribute(userId, schemaType, schemaTypeGUID, attribute, attribute.getDataType(), externalSourceName);
             } else {
                 String schemaAttributeGUID = schemaAttributeEntity.get().getGUID();
-                EntityDetail updatedSchemaAttributeEntity = buildSchemaAttributeEntityDetail(schemaAttributeGUID, schemaAttribute);
+                EntityDetail updatedSchemaAttributeEntity = buildSchemaAttributeEntityDetail(schemaAttributeGUID, attribute);
                 EntityDetailDifferences entityDetailDifferences = repositoryHelper.getEntityDetailDifferences(schemaAttributeEntity.get(),
                         updatedSchemaAttributeEntity, true);
 
                 if (entityDetailDifferences.hasInstancePropertiesDifferences()) {
-                    schemaTypeHandler.updateSchemaAttribute(userId, schemaAttributeGUID, schemaAttribute);
+                    String externalSourceGUID = dataEngineRegistrationHandler.getExternalDataEngineByQualifiedName(userId, externalSourceName);
+                    schemaAttributeHandler.updateSchemaAttribute(userId,
+                            externalSourceGUID,
+                            externalSourceName,
+                            schemaAttributeGUID,
+                            getSchemaAttributeBuilder(attribute).getInstanceProperties(methodName));
                 }
             }
         }
     }
 
-    private EntityDetail buildSchemaAttributeEntityDetail(String schemaAttributeGUID, SchemaAttribute schemaAttribute) throws
-                                                                                                                       InvalidParameterException {
+
+    private EntityDetail buildSchemaAttributeEntityDetail(String schemaAttributeGUID, Attribute attribute) throws
+            InvalidParameterException {
         String methodName = "buildSchemaAttributeEntityDetail";
+        SchemaAttributeBuilder schemaAttributeBuilder = getSchemaAttributeBuilder(attribute);
 
-        SchemaAttributeBuilder builder = schemaTypeHandler.getSchemaAttributeBuilder(schemaAttribute);
+        return dataEngineCommonHandler.buildEntityDetail(schemaAttributeGUID, schemaAttributeBuilder.getInstanceProperties(methodName));
+    }
 
-        return dataEngineCommonHandler.buildEntityDetail(schemaAttributeGUID, builder.getInstanceProperties(methodName));
+    SchemaAttributeBuilder getSchemaAttributeBuilder(Attribute attribute) {
```

Comment:
minCardinality. maxCardinality, defaultValueOverride, allowsDuplicateValues, orderedValues need to be set from Attribute bean